src/gen/include/RAT/LiGen.hh

```diff
@@ -0,0 +1,76 @@
+//fsutanto@umich.edu
+//akindele1@llnl.gov
+//Jan 6, 2020
+
+// To use this generator, the command is:
+// /generator/add li ISOTOPE:POSITION[:TIME]
+// For example:
+// /generator/add li 9:fill
+
+#ifndef __RAT_LiGen__
+#define __RAT_LiGen__
+
+#include <RAT/GLG4Gen.hh>
+#include <globals.hh>
+#include <Randomize.hh>
+
+class G4Event;
+class G4ParticleDefinition;
+class GLG4TimeGen;
+class GLG4PosGen;
+
+namespace RAT {
+    
+    class LiGen : public GLG4Gen {
+    public:
+        LiGen();
+        virtual ~LiGen();
+        virtual void GenerateEvent(G4Event *event);
+        virtual void ResetTime(double offset=0.0);
+        virtual bool IsRepeatable() const { return true; };
+        
+        virtual void SetState(G4String state);
+        virtual G4String GetState() const;
+        
+        virtual void SetTimeState(G4String state);
+        virtual G4String GetTimeState() const;
+        virtual void SetPosState(G4String state);
+        virtual G4String GetPosState() const;
+        
+    protected:
+        
+        // Generator initialization, specified by the user.
+        G4String stateStr;
+        
+        // isotope.
+        int isotope;
+        
+        // The time and position generators specified by the user.
+        GLG4TimeGen* timeGen;
+        GLG4PosGen* posGen;
+        
+        // The Li9Source event model only generate neutrons and photons.
+        G4ParticleDefinition* neutron;
+        G4ParticleDefinition* electron;
+        
+    private:
+        
+        G4RandGeneral* spectrumSampler;
+        virtual void SetUpBetaSpectrumSampler( G4double& e0 );
+        virtual G4double FermiFunction( G4double& W);
+        virtual G4double ModSquared( G4double& re, G4double& im);
+        
+        G4int Z;
+        G4int A;
+        G4double alphaZ;
+        G4double Rnuc;
+        G4double V0;
+        G4double gamma0;
+        G4double sumBr;
+        G4double pdfNow;
+        
+    };
+    
+} // namespace RAT
+
+#endif // RAT_LiGen_h
```

Comment:
@feliciasutanto Mind fixing up the whitespace in this file so that it isn't adding extra indentation throughout?

Reply:
Sorry, not this file. I mean Gsim.cc, though maybe fix this one to be consistent with the others. Not that I agree with rat-pac's choice to not indent inside the scope of a namespace, but it would be better to be consistent than correct

Reply:
Sure, I deleted the indentation.

Reply:
Ah wait, I have a typo.... I will push something very soon. Don't merge it yet..

Reply:
Ok, fixed. It compiles fine now.